Feature/#43 - Docker Execution Environment added (#52)
courselab/snaskii21: docker_env/README.md

### TTSnake Docker execution environment 
### Author: @Henrique1792
### Date: 2021-11-15

1. Rationale
	After many installation workarounds pointed by colleagues, this suggestion

> 	was created in order to easy up everyone on system dependencies.

Reviewer: Pretty sure the expression is "ease up" and not "easy up". 

```suggestion
	was created in order to ease up everyone on system dependencies.
```

	It intends to allow execution without bloating host environment and solve
	compatibility issues during execution.

2. Usage
	1. Get into `docker_env` directory and build it executing:
		`docker build -t snake_env .`

> 	2. After it's build, run container as follows (newbie's gentle execution):
> 	`docker run -it --rm -v <snaskii21_path>:/data -w /data snake_env /bin/bash`
> 	
> 		Rationale:
> 			`-it` stands for `iterative` mode (-i keeps STDIN attached and open, while -t allocate a pseudo-tty)
> 			`--rm` cleans container up after execution (for small tests this flag is way useful to ignore)
> 			`-v HOST_DIR:CONTAINER_DIR` sets a host directory share point inside the container
> 			`-w WORKDIR_NAME` is your initial working directory after bring it up

Reviewer: Found typo on line 105.
Also I thought line 112 was quite unclear and removed the last part of it (that can be just my opinion so I'm open to discussion 😛).

```suggestion
	2. After it's built, run container as follows (newbie's gentle execution):
	`docker run -it --rm -v <snaskii21_path>:/data -w /data snake_env /bin/bash`
	
		Rationale:
			`-it` stands for `iterative` mode (-i keeps STDIN attached and open, while -t allocate a pseudo-tty)
			`--rm` cleans container up after execution (for small tests this flag is way useful to ignore)
			`-v HOST_DIR:CONTAINER_DIR` sets a host directory share point inside the container
			`-w WORKDIR_NAME` is your initial working directory
```

> 3. Improvements
> 	1. This container gets too big if the environment suffer any modification. Maybe reconsider some steps and
> 		keep just execution requirements
> 	2. **Advices for compilation**
> 		You may log as super user during all process inside docker (sudo configuration isn't really recommended);
> 		But compilation steps which doesn't require that much permission as snakett user. Switch it running
> 		`su snakett` command - if root is required again, CTRL-D will get you back (or exit command)
> 	3. The author isn't familiar with Github actions structure, but I understand that we can create a deploy
> 		img from that structure, which I hope we're able to use =) .

Reviewer: I didn't quite get this section. Is this about ways we could improve these Docker instructions? If so I think we should leave this as comment so common users won't see it (which I made in my suggestion) or maybe make the text cleaner, matching the rest of the document.

```suggestion
<!--3. Improvements
	1. This container gets too big if the environment suffer any modification. Maybe reconsider some steps and
		keep just execution requirements
	2. **Advices for compilation**
		You may log as super user during all process inside docker (sudo configuration isn't really recommended);
		But compilation steps which doesn't require that much permission as snakett user. Switch it running
		`su snakett` command - if root is required again, CTRL-D will get you back (or exit command)
	3. The author isn't familiar with Github actions structure, but I understand that we can create a deploy
		img from that structure, which I hope we're able to use =) .-->
```

Author: Last section is about structures which could be applied - also, root permission for building could be harmfull for an unexperienced user; yet I agree with keeping it as a commentary - this considerations might be helpful but doesn't act like any sort of requirement. Thank you!
